### ISSUE TITLE
TCP6ClientEndpoint._deferToThread should be a free-function and not a bound method

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6633]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)| @ashfall reported|
|-|-|
|Trac ID|trac#6633|
|Type|defect|
|Created|2013-07-18 20:23:37Z|
|Branch|https://github.com/twisted/twisted/tree/free-function-TCP6ClientEndpoint-deferToThread-6633|

TCP6ClientEndpoint fails to work because of this:

```
>>> e = endpoints.TCP6ClientEndpoint(reactor, "www.google.com", 80)
>>> e.connect(Factory.forProtocol(Protocol))
<Deferred #0>
Deferred #0 failed: "'TCP6ClientEndpoint' object is not callable"
```

<details><summary>Searchable metadata</summary>

```
trac-id__6633 6633
type__defect defect
reporter__ashfall ashfall
priority__normal normal
milestone__ 
branch__branches_free_function_TCP6ClientEndpoint_deferToThread_6633 branches/free-function-TCP6ClientEndpoint-deferToThread-6633
branch_author__ashfall ashfall
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1374179017000000 1374179017000000
changetime__1374954109000000 1374954109000000
version__None None
owner__ashfall ashfall

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_1"></a>|@ashfall commented|
|-|-|

(In [39235]) Branching to 'free-function-TCP6ClientEndpoint-deferToThread-6633'

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)|@ashfall removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_3"></a>|@ashfall commented|
|-|-|

Forced builds.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @ashfall|
|-|-|

Thanks!  I think the news fragment could be a little more clear.  The user-visible consequence of this change is that `TCP6ClientEndpoint` now works when constructed with a hostname - whereas previously it could not establish any connections in that case.  That's all.  Please merge after fixing that.

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_5"></a>|@ashfall set status to `closed`|
|-|-|

(In [39341]) Merge free-function-TCP6ClientEndpoint-deferToThread-6633: Make TCP6ClientEndpoint._deferToThread a free-function instead of a bound method.

Author: ashfall
Reviewer: exarkun
Fixes: [#6633](https://github.com/twisted/twisted/issues/6633)
